test/controllers/changeset_comments_controller_test.rb

```diff
@@ -46,6 +46,14 @@ def test_feed
         assert_select "item", :count => 3
       end
     end
+
+    # Gets comment Ids from HTML and checks that they are in descending order
+    last_comment_id = -1
+    @response.body.split("Changeset comment ")[1..].each_with_index do |data, index|
```

Comment:
It would be better to test against the parsed response rather than trying to use string operations to inspect HTML embedded in feed XML from the raw response.

What I'm not sure about is how easy that is when you're looking at the HTML content.

I think we should certainly be able to use he parsed response to find the feed entries though even if we then extract the ID from the HTML using string operations.

Reply:
Thanks for comments. I agree with you, but when I parsed XML, the only id info was still in the CDATA containing HTML (fields exposed in the XML were identical for every comment e.g. comment creation time, because they are created with no delay). So even if I used parsed data, I would still use HTML raw string manipulation. Certainly, it's doable (and at first I did it that way), but it makes the test more complicated (so I rewrote it this way). If it's okay, test to be more complicated, I'll do it with XML parser, but I think this solution is already more complicated than the regular test should be.

Reply:
I think there is a test somewhere that parses embedded html.

Reply:
I changed the test to use parsed XML and HTML parser.

Reply:
This is the test I meant
https://github.com/openstreetmap/openstreetmap-website/blob/e3c919275818d46a515c183cf373d7a53bc6942a/test/controllers/traces_controller_test.rb#L729